### PR TITLE
Add CAN support for FRDM-K64F board

### PR DESCRIPTION
**Pull request by alexanderwachter**
This PR enables CAN for the FRDM-K64F board.

### CODE REVIEW

**Review by dbkinder — APPROVED** [review]
docs LGTM, thanks.

**Review by dleach02 — APPROVED** [review]

File: drivers/can/can_mcux_flexcan.c [hunk 1]
```diff
@@ -530,11 +530,29 @@ DEVICE_AND_API_INIT(can_mcux_flexcan_0, DT_INST_0_NXP_KINETIS_FLEXCAN_LABEL,
 
 static void mcux_flexcan_config_func_0(struct device *dev)
 {
+#if defined(CONFIG_SOC_SERIES_KINETIS_K6X)
```
Comment (henrikbrixandersen) [inline drivers/can/can_mcux_flexcan.c]: I suggest `#ifdef`ing on whether or not the specific IRQ is defined here - that makes for a more flexible driver.

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
Looks good! It's nice to see a driver that was merged only a few days ago already finding new uses :-)

File: drivers/can/can_mcux_flexcan.c (hunk 1 above)
Comment (alexanderwachter) [reply drivers/can/can_mcux_flexcan.c]: Done.

**Comment by alexanderwachter** [comment]
@MaureenHelm  @henrikbrixandersen already asked for a config flag for the LOM bit.
Additionally we need to set the SRXDIS bit to get the same behavior as the STM32 CAN controller.

**Review by MaureenHelm — APPROVED** [review]

**Review by henrikbrixandersen — APPROVED** [review]
Looks good. I have verified the changed on the TWR-KE18F board.

**Pull request merged**